File: MTWDM iOS Xamarin/App12/App12/ViewController.cs
using System;
using System.Linq;
using Foundation;
using UIKit;

namespace App12
{
    public partial class ViewController : UITableViewController
    {
        //Control para actualizar datos de nuestra TableView
        UIRefreshControl pullToRefreshControl = new UIRefreshControl();

        Modelo modelo = Modelo.GetInstance();

        protected ViewController(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
            // Perform any additional setup after loading the view, typically from a nib.
            this.NavigationItem.LeftBarButtonItem = this.EditButtonItem;
            this.NavigationItem.RightBarButtonItem = new UIBarButtonItem(UIBarButtonSystemItem.Add,
                                                                         (sender, e) => nuevoComestible());

            this.RefreshControl = pullToRefreshControl;
            pullToRefreshControl.AddTarget((sender, args) => refreshTable(), UIControlEvent.ValueChanged);
        }

        public override void ViewWillAppear(bool animated)
        {
            TableView.ReloadData();
        }

        public void nuevoComestible()
        {
            PerformSegue("segueNuevoComestible", this);
        }


        public void refreshTable()
        {
            modelo.datos = modelo.datosNube;
            TableView.ReloadData();
            pullToRefreshControl.EndRefreshing();
        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }

        public override nint NumberOfSections(UITableView tableView)
        {
            return 1;
        }

        public override nint RowsInSection(UITableView tableView, nint section)
        {
            return modelo.datos.Length;
        }

        public override UITableViewCell GetCell(UITableView tableView, NSIndexPath indexPath)
        {

            var cell = tableView.DequeueReusableCell("comestiblesCell", indexPath);

            if (cell == null)
            {
                cell = new UITableViewCell(UITableViewCellStyle.Default, "comestiblesCell");
            }

            cell.TextLabel.Text = modelo.datos[indexPath.Row];

            return cell;

        }

        public override bool CanEditRow(UITableView tableView, NSIndexPath indexPath)
        {
            return true;
        }

        public override void CommitEditingStyle(UITableView tableView, UITableViewCellEditingStyle editingStyle, NSIndexPath indexPath)
        {

            if (editingStyle == UITableViewCellEditingStyle.Delete)
            {
                var datos = modelo.datos.ToList();
                datos.RemoveAt(indexPath.Row);

                modelo.datos = datos.ToArray();

                TableView.DeleteRows(new NSIndexPath[] { indexPath },
                                     UITableViewRowAnimation.Middle);

            }
            else if (editingStyle == UITableViewCellEditingStyle.Insert)
            {

            }
        }

        public override bool CanMoveRow(UITableView tableView, NSIndexPath indexPath)
        {
            return true;
        }


        public override void MoveRow(UITableView tableView, NSIndexPath sourceIndexPath, NSIndexPath destinationIndexPath)
        {
            //inserccion, afectacion a una api
        }
    }
}

File: MTWDM iOS Swift /app11/app11/ViewController.swift
//
//  ViewController.swift
//  app11
//
//  Created by Felipe Hernandez on 10/19/18.
//  Copyright © 2018 Felipe Hernandez. All rights reserved.
//

import UIKit

class ViewController: UITableViewController {

    // MARK: - View Did Load
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.navigationItem.leftBarButtonItem = self.editButtonItem
        
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(segueToNewGrocery))
        
        self.refreshControl = pullToRefreshControl
        
        pullToRefreshControl.addTarget(self, action: #selector(refreshTable), for: .valueChanged)
    }
    
    // MARK: - Properties
    let pullToRefreshControl = UIRefreshControl()
    let modelo = Modelo()
    
    // MARK: - Functions
    @objc func refreshTable() {
        modelo.datos = modelo.datosNube
        self.tableView.reloadData()
        pullToRefreshControl.endRefreshing()
    }
    
    @objc func segueToNewGrocery() {
        
    }
    
    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return modelo.datos.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCell(withIdentifier: "comestiblesCell", for: indexPath)
    
        cell.textLabel?.text = modelo.datos[indexPath.row]
        
        return cell
    }
    
    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        return true
    }
    
    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
        
        if editingStyle == .delete {
            modelo.datos.remove(at: indexPath.row)
            tableView.deleteRows(at: [indexPath], with: .middle)
        } else if editingStyle == .insert {
        
        }
    }
    
    override func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {
        //inserccion, afectacion a una api etc...
    }
    
    override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
        return true
    }
}

